fix: GO compiler names

dkm opened the pull request:
GO compiler naming was not coherent (and some were incorrect). Try to make them more homogeneous.

fixes #4733

dkm commented:
@Alcaro, does that look better? Any suggestion? It's painful to test locally, so the real test will be done in staging, but you can already comment by looking at the change :)

Alcaro commented:
Inconsistent capitalization

https://github.com/compiler-explorer/compiler-explorer/pull/4746/files#diff-0d845469ccad7c8374beb46032d4647031520cc3250f415c61466d5f9c4a9b51R388
https://github.com/compiler-explorer/compiler-explorer/pull/4746/files#diff-0d845469ccad7c8374beb46032d4647031520cc3250f415c61466d5f9c4a9b51R400

But category names are text-transform:uppercase anyways, so that doesn't matter.

I don't know if those name/baseName/etc fields actually work, nor do I know if it puts i386 beside x86_64, but if this change does what it looks like it does, looks good to me.

dkm commented:
Thanks. I think I've fixed all the case-inconsistencies. I've moved around the group/baseNames to easily check them.
Not sure we can force x86* to be altogether. IIRC, there's some alpha sort somewhere, so "maybe" this will work.

mattgodbolt reviewed (approved):
Passes the "it won't crash in prod" test :) Thanks!
